### ISSUE TITLE
Add Support for transactions to our SurrealDB layer

### ISSUE THREAD

**Issue by zupzup** [BitcreditProtocol/Bitcredit-Core#278]
Because we have plenty of cases, where we want to do transactionally atomic operations (e.g. creating a bill also creates a bill chain, a company chain entry and an identity chain entry) and we want to automatically roll back, if one of them fails.

Unfortunately, the Rust SDK doesn't support this and in the ticket, there doesn't seem to be any progress. (https://github.com/surrealdb/surrealdb/issues/2733)

So we need to find a solution on our side. For example, that we have "direct" operations (as we have now) and "transactional" functions of these operations, which simply return the query for the operation, so multiple queries from multiple operations can be combined to one transactional function call, which uses "begin transaction" and "end transaction" and adds all collected queries between.

**Comment by mubarak23** [comment]
is this issue related to this [PR](https://github.com/BitcreditProtocol/E-Bill/pull/279) @zupzup

**Comment by zupzup** [comment]
> is this issue related to this [PR](https://github.com/BitcreditProtocol/E-Bill/pull/279) @zupzup

Yes, it's related, but it was labelled wrong. We'll have to decide on an API for it first, before work can start.

**Comment by zupzup** [comment]
So far, this never happened

**Comment by zupzup** [comment]
never happened
won't use surreal in the future
